packages/live-preview-sdk/src/helpers/validation.ts

```diff
@@ -22,10 +33,34 @@ function validation(d: Argument): { isGQL: boolean; sysId: string | null; isREST
     }
 
     // maybe it's nested
-    return validation(Object.values(d) as Argument);
+    return validateData(Object.values(d) as Argument, maxDepth - 1);
   }
 }
 
+function validatedConfig(originalConfig: ContentfulSubscribeConfig, isREST: boolean) {
+  const config = { ...originalConfig };
+
+  if (config.query) {
+    if (typeof config.query !== 'object') {
```

Comment:
would we consider parsing it in future?

Reply:
that would increase the bundle size of our package as we need graphql-tag as a dependency (at least peerDependency)
We could try with optional dependency maybe, can give it a try tomorrow

Reply:
this is kind of interesting: https://docs.npmjs.com/cli/v7/configuring-npm/package-json#peerdependenciesmeta

Reply:
graphql-tag is 2kb, so I imported it directly now
Otherwise we would rework the whole subscriptions to be async :(